### PR TITLE
Make the item property completely customizable

### PR DESCRIPTION
**Pull request by syndbg**
Allow to pass React components.

This will make it extremely easy to use this component with literally everything.

From list views, to menus, to tables and what not. My use case is to use it for a menu, but you get the idea.

Also I've added back the `dist` folder. It's completely common and fine for NPM projects to have the `dist` folder in their repos. 

This allows repo contributors to test their packages easily by installing the npm package from a github repo url. Currently this is not possible due to https://github.com/cht8687/react-expandable-listview/blob/master/package.json#L5 never existing.

### CODE REVIEW

**Comment by cht8687** [comment]
thank you so much @syndbg , I will review the changes today and this looks great!!!

**Comment by cht8687** [comment]
@stevemao, thoughts about this PR?

File: src/lib/ListItem.js [hunk 1]
```diff
@@ -1,14 +1,18 @@
-import React, { Component, PropTypes } from 'react'
+import React from 'react'
 
-export default class ListItem extends Component {
-  static propTypes = {
-    item: PropTypes.string.isRequired
+export default class ListItem extends React.Component {
+  static get propTypes() {
+    return {
+      children: React.PropTypes.object,
```
Comment (stevemao) [inline src/lib/ListItem.js:6]: Can you provide some reference why this pattern is better?

File: .gitignore [hunk 2]
```diff
@@ -7,8 +7,6 @@ pids
 *.pid
 *.seed
 
-/dist/
```
Comment (stevemao) [inline .gitignore:10]: I'd prefer to add the build script to `npm install` so that it will build automatically.

**Comment by stevemao** [comment]
Otherwise LGTM. This is something I'd always wanted to see :)

File: src/lib/ListItem.js (hunk 1 above)
Comment (syndbg) [reply src/lib/ListItem.js:6]: Unlimited freedom of what you can do with the `react-expandable-listview`.

I can add my own custom elements as items and let `react-expandable-listview` manage their state (whether they're being open and etc).

In a current project I use `react-expandable-listview` combined with `react-router` `Link` elements to make an admin menu that's expandable.

E.g

``` JavaScript
export default class Menu extends React.Component {
  static get menuItems() {
    return [
      {
        headerName: 'Products',
        isOpened: false,
        height: 100,
        items: [
          (
          <Link
            to="admin/products/all"
            className="btn btn-default"
            activeClassName="active"
          >
            All
          </Link>
          ),
          (
          <Link
            to="admin/products/expired"
            className="btn btn-default"
            activeClassName="active"
          >
            Expired
          </Link>
          ),
          (
          <Link
            to="admin/products/submitted"
            className="btn btn-default"
            activeClassName="active"
          >
            Submitted
          </Link>
          ),
        ],
      },
      {
        headerName: 'Promotions',
        isOpened: false,
        height: 100,
        items: [
          (
          <Link
            to="admin/promotions/active"
            className="btn btn-default"
            activeClassName="active"
          >
            Active
          </Link>
          ),
       ],
      },
      {
        headerName: 'Settings',
        isOpened: false,
        height: 100,
        items: [
          (
          <Link
            to="admin/settings/all"
            className="btn btn-default"
            activeClassName="active"
          >
            Al
          </Link>
          ),
        ],
      },
    ];
  }

  render() {
    return (
      <div id="admin-menu">
        <ReactExpandableListView
          data={this.constructor.menuItems}
          headerAttName="headerName"
          itemsAttName="items"
        />
      </div>
    );
  }
}
```

File: .gitignore (hunk 2 above)
Comment (syndbg) [reply .gitignore:10]: This is the current behavior.

Without changing `package.json` a `npm install` invocation triggers `prepublish` which does automatically build the distributed version (ref: https://github.com/cht8687/react-expandable-listview/blob/master/package.json#L8)

**Comment by cht8687** [comment]
@syndbg 
image

I pulled your branch locally, and when I do webpack-dev-server, I saw there is an error in the console above. 
I am going to further investigate it. Probably due to data structure changed. 

Once I figured it out, will merge this awesome PR...!!

File: .gitignore (hunk 2 above)
Comment (stevemao) [reply .gitignore:10]: Yeah, I think adding `npm run prepublish` to `npm install` is better.

**Comment by cht8687** [comment]
@syndbg  I am going to take this PR, but I will add some options for rendering plain object... as currently it is throwing out errors when try to render plain object.

**Pull request merged**